celery/app/autoretry.py

```diff
@@ -1,35 +1,56 @@
 """Tasks auto-retry functionality."""
+import inspect
+
 from vine.utils import wraps
 
 from celery.exceptions import Ignore, Retry
 from celery.utils.time import get_exponential_backoff_interval
 
 
+def check_reserved_arg_availability(fun):
+    arg_spec = inspect.getfullargspec(fun)
+    if 'autoretry_opts' in set(arg_spec.args):
+        raise ValueError(
+            "'autoretry_opts' is a reserved argument to override "
+            "autoretry options. Use a different name for this "
+            "parameter please."
+        )
+
+
 def add_autoretry_behaviour(task, **options):
     """Wrap task's `run` method with auto-retry functionality."""
-    autoretry_for = tuple(
+    check_reserved_arg_availability(task._orig_run)
+
+    _autoretry_for = tuple(
         options.get('autoretry_for',
                     getattr(task, 'autoretry_for', ()))
     )
-    retry_kwargs = options.get(
+    _retry_kwargs = options.get(
         'retry_kwargs', getattr(task, 'retry_kwargs', {})
     )
-    retry_backoff = int(
+    _retry_backoff = int(
         options.get('retry_backoff',
                     getattr(task, 'retry_backoff', False))
     )
-    retry_backoff_max = int(
+    _retry_backoff_max = int(
         options.get('retry_backoff_max',
                     getattr(task, 'retry_backoff_max', 600))
     )
-    retry_jitter = options.get(
+    _retry_jitter = options.get(
         'retry_jitter', getattr(task, 'retry_jitter', True)
     )
 
-    if autoretry_for and not hasattr(task, '_orig_run'):
+    if _autoretry_for and not hasattr(task, '_orig_run'):
 
         @wraps(task.run)
-        def run(*args, **kwargs):
+        def run(*args, autoretry_opts=None, **kwargs):
+            if autoretry_opts is None:
```

Comment:
if autoretry_opts is None: or if autoretry_opts is not None:?

Reply:
`if autoretry_opts is None`, it's just to not set `{}` as default argument because it's mutable. 

Even if it would be safe to do in this case it's better to avoid to do so.